### PR TITLE
feat(env): adds a list of supported env vars and a generator for such list.

### PR DESCRIPTION
**Pull request by jcchavezs**
This PR adds support for a list of environment variables along with a tool for generating them directly from the proto.

Ping @hypertrace/agent-maintainers

### CODE REVIEW

File: tools/env-vars-generator/main.go [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+		newTemplate := []byte{}
+		newTemplate = append(newTemplate, template[:commentIdx]...)
+		newTemplate = append(newTemplate, template[newLineIdex+1:]...)
+		template = newTemplate
+	}
+
+	return template, nil
+}
+
+// readTypes read all types in the proto file to be able to list them and identify those who are
```
Comment (jcchavezs) [inline tools/env-vars-generator/main.go:144]:
```suggestion
// getTypes read all types in the proto file to be able to list them and identify those who are
```

File: .gitmodules [hunk 2]
```diff
@@ -0,0 +1,3 @@
+[submodule "tools/env-vars-generator/protobuf"]
```
Comment (jcchavezs) [inline .gitmodules:1]: We need this for the parser to be able to resolve scalar wrappers in proto.

File: config.proto [hunk 3]
```diff
@@ -31,7 +30,7 @@ message Reporting {
     // address represents the host for reporting the traces e.g. api.traceable.ai
     google.protobuf.StringValue address = 1;
 
-    // secure when false, permits connecting to the trace endpoint without a certificate
+    // when `true`, avoids connecting to the trace endpoint without a certificate
```
Comment (davexroth) [inline config.proto:33]: this comment is a little hard to read.  maybe:
when `true`, connects to endpoints over tls.

**Review by davexroth — APPROVED** [review]

File: config.proto (hunk 3 above)
Comment (jcchavezs) [inline config.proto:33]:
```suggestion
    // when `true`, it connects to endpoints over TLS.
```

Comment (jcchavezs) [reply config.proto]: Done.

Comment (mohit-a21) [reply config.proto:35]: Is this getting reflected in `ENV_VARS.md`?

**Review by mohit-a21 — CHANGES_REQUESTED** [review]

File: config.proto (hunk 3 above)
Comment (jcchavezs) [reply config.proto:35]: Done.

**Review by mohit-a21 — APPROVED** [review]

**Pull request merged**